{{cookiecutter.github_project_name}}/js/.npmignore

```diff
@@ -0,0 +1 @@
+{{ cookiecutter.npm_package_name }}-*.tgz
```

Comment:
I would not add an `.npmignore` if it doesn't include anything that is matched by the .gitignore file.

Reply:
the `package.json` file and the `.gitignore` files are not at the same level, so I would not expect npm to ever take into account the `.gitignore` file, or does it?

Reply:
Should be easy to try, but I would assume it does?

Reply:
I tried locally. And indeed it does not take the `.gitignore` into account when running `npm` from the `js` dir, which makes sense IMO. Of course it might be different when running `npm` from the root directory of the project, which I usually don't do myself, but maybe your workflow is different?

Reply:
I think npm is smart about how it finds the .gitignore, and given how many packages rely on this fact, I think our cookiecutter can as well ;)

Reply:
Sorry, I am not sure I get your point here, what do you think we should do then?

`npm` does not seem smart enough to take the `.gitignore` from the parent directory when run from the `js` directory, so we cannot really rely on it.

Reply:
I'm sorry, I missed the "not" in your previous comment. Kind of disappointing :-/

Reply:
Indeed it's disappointing. It would not be a problem in the ts-cookiecutter though ;)